enhancements/ingress/aws-load-balancer-operator.md

```diff
@@ -0,0 +1,394 @@
+    http:
+        paths:
+          - path: /
+            pathType: Exact
+            backend:
+              service:
+                name: example-service
+                port:
+                  number: 80
+```
+The controller then creates an application load balancer with an HTTPS listener
+which terminates the TLS connection with the certificate ARN which is specified
+in the annotation. If the certificate does not exist, the load balancer is not
+created. It also enables AWS Shield on the Elastic IP associated with the ALB.
+
+#### Multiple Ingress through single ALB
+
+The user wants to route traffic to services example-1, example-2 and example-3
+as parts of the domain example.com and wants all the endpoints served through a
+single ALB. The user create an `AWSLoadBalancerController` like in the previous example and
+then creates 3 Ingress resources as follows:
+
+```yaml
+kind: Ingress
+metadata:
+  name: example-1
+  annotations:
+    alb.ingress.kubernetes.io/scheme: internet-facing
+    alb.ingress.kubernetes.io/group.name: example
+    alb.ingress.kubernetes.io/group.order: “1”
+spec:
+  ingressClass: alb
+  rules:
+  - host: example.com
+    http:
+        paths:
+        - path: /blog
+          backend:
+            service:
+        name: example-1
+        port:
+          number: 80
+—--
+kind: Ingress
+metadata:
+  name: example-2
+  annotations:
+    alb.ingress.kubernetes.io/scheme: internet-facing
+    alb.ingress.kubernetes.io/group.name: example
+    alb.ingress.kubernetes.io/group.order: “2”
+spec:
+  ingressClass: alb
+  rules:
+  - host: example.com
+    http:
+        paths:
+        - path: /store
+          backend:
+            service:
+        name: example-2
+        port:
+          number: 80
+—--
+kind: Ingress
+metadata:
+  name: example-3
+  annotations:
+    alb.ingress.kubernetes.io/scheme: internet-facing
+    alb.ingress.kubernetes.io/group.name: example
+    alb.ingress.kubernetes.io/group.order: “3”
+spec:
+  ingressClass: alb
+  rules:
+  - host: example.com
+    http:
+        paths:
+        - path: /
+          backend:
+            service:
+        name: example-3
+        port:
+          number: 80
+```
+
+In this example the annotations `alb.ingress.kubernetes.io/group.name` specifies
+which Ingress Group the Ingresses belong to and the annotation
+`alb.ingress.kubernetes.io/group.order` specifies the order in the group which
+will be used for path matching. All Ingresses which belong to the same group are
+attached to the same ALB.
+
+### API Extensions
+
+This operator uses a cluster-scoped Custom Resource to configure the
+lb-controller instance. The resource has the following _Spec_:
+
+```golang
+type AWSLoadBalancerControllerSpec struct {
+  AutoTagSubnet bool // indicates if the operator should tag the subnets
+  // +optional
+  AdditionalResourceTags map[string]string // Default AWS Tags that will be applied to all AWS resources managed by this controller (default [])
```

Comment:
Are tags specified in OpenShift's cluster infrastructure config used at all by this operator?

Reply:
For the initial version of the operator it will not use the tags specified in the `Infrastructure` resource. Users will have to specify it explicitly.